apps/extension/src/core/views/ActivityItem.tsx

```diff
@@ -1,13 +1,61 @@
+import {
+  MediaMimeType,
+  isMediaOutput,
+  isPromptInput,
+  mimeTypeToExtension,
+  isMediaMimeType
+} from "window.ai"
+
 import { Logo } from "~core/components/pure/Logo"
 import { Text } from "~core/components/pure/Text"
 import { originManager } from "~core/managers/origin"
 import type { Transaction } from "~core/managers/transaction"
 import { transactionManager } from "~core/managers/transaction"
 import { formatDate } from "~core/utils/utils"
 
+function createMediaDownloadLinks(transaction: Transaction) {
+  if (!transaction.input || !transaction.outputs) {
+    return null
+  }
+  const input = transaction.input
+  // fallback to .txt containing contents
+  let mimeType = MediaMimeType.TXT
+  const outputs = transaction.outputs
+  if (!isPromptInput(input) || !outputs.every(isMediaOutput)) {
+    return null
+  }
+  //check if a valid mimetype is contained in the data uri
+  const mimeTypeFromUri = outputs[0].uri.split(";")[0].split(":")[1]
+  if (isMediaMimeType(mimeTypeFromUri)) {
+    mimeType = mimeTypeFromUri
+  }
+
+  return outputs.map((output, index) => {
+    const fileName = `${input.prompt.replace(" ", "-")}_${index}_${mimeTypeToExtension[mimeType]}`
+    return (
+      <>
+        <br></br>
```

Comment:
usually this is a self-closing tag: `<br />`

Reply:
rather than separating with `<br />`, easier/simpler to just have `<a>` tags that have "block" as a classname, to put them on new lines

Reply:
changed to "block"